imap_processing/ultra/l0/decom_ultra.py

```diff
@@ -1,211 +1,139 @@
-                break
-
-            value, current_position = read_n_bits(binary, width, current_position)
-
-            # Log decompression
-            decompressed_values.append(log_decompression(value))
-
-    return decompressed_values
-
-
-def decom_ultra_packets(packet_file: str, xtce: str):
-    """
-    Unpack and decode ultra packets using CCSDS format and XTCE packet definitions.
+    Unpack and decode Ultra packets using CCSDS format and XTCE packet definitions.
 
     Parameters
     ----------
     packet_file : str
         Path to the CCSDS data packet file.
     xtce : str
         Path to the XTCE packet definition file.
+    test_apid : Optional[int]
+        The APID to test. If None, all APIDs are processed.
 
     Returns
     -------
-    xr.Dataset
-        A dataset containing the decoded data fields with 'time' as the coordinating
-        dimension.
+    decom_data : dict
+        A dictionary containing the decoded data.
     """
     packets = decom.decom_packets(packet_file, xtce)
-
-    met_data, science_id, spin_data, abortflag_data, startdelay_data, fastdata_00 = (
-        [],
-        [],
-        [],
-        [],
-        [],
-        [],
-    )
-
-    for packet in packets:
-        if (
-            packet.header["PKT_APID"].derived_value
-            == UltraParams.ULTRA_IMG_RATES.value.apid
+    grouped_data = group_by_apid(packets)
+    decom_data = {}
+
+    if test_apid:
+        grouped_data = {test_apid: grouped_data[test_apid]}
+
+    for apid in grouped_data.keys():
+        if not any(
+            apid in category.value.apid
+            for category in [
+                UltraParams.ULTRA_EVENTS,
+                UltraParams.ULTRA_AUX,
+                UltraParams.ULTRA_TOF,
+                UltraParams.ULTRA_RATES,
+            ]
         ):
-            met_data.append(packet.data["SHCOARSE"].derived_value)
-            science_id.append(packet.data["SID"].derived_value)
-            spin_data.append(packet.data["SPIN"].derived_value)
-            abortflag_data.append(packet.data["ABORTFLAG"].derived_value)
-            startdelay_data.append(packet.data["STARTDELAY"].derived_value)
-            decompressed_data = decompress_binary(
-                packet.data["FASTDATA_00"].raw_value,
-                UltraParams.ULTRA_IMG_RATES.value.width,
-                UltraParams.ULTRA_IMG_RATES.value.block,
-            )
-            fastdata_00.append(decompressed_data)
-
-    array_data = np.array(fastdata_00)
-
-    ds = xr.Dataset(
-        {
-            "science_id": ("epoch", science_id),
-            "spin_data": ("epoch", spin_data),
-            "abortflag_data": ("epoch", abortflag_data),
-            "startdelay_data": ("epoch", startdelay_data),
-            "fastdata_00": (("epoch", "index"), array_data),
-        },
-        coords={
-            "epoch": met_data,
-        },
-    )
-
-    return ds
+            logging.info(f"{apid} is currently not supported")
```

Comment:
Use the logger for this module. Do you want to warn on this instead of info?
```suggestion
            logger.info(f"{apid} is currently not supported")
```